sdk/src/test/scala/com/horizen/account/performance/AccountSidechainNodeViewHolderPerfTest.scala

```diff
@@ -0,0 +1,601 @@
+
+import akka.actor.{ActorRef, ActorSystem, Props}
+import akka.testkit.TestActorRef
+import com.horizen.account.AccountSidechainNodeViewHolder
+import com.horizen.account.block.AccountBlock
+import com.horizen.account.fixtures.EthereumTransactionFixture
+import com.horizen.account.history.AccountHistory
+import com.horizen.account.mempool.AccountMemoryPool
+import com.horizen.account.state.{AccountState, AccountStateView, MessageProcessor}
+import com.horizen.account.storage.{AccountHistoryStorage, AccountStateMetadataStorage}
+import com.horizen.account.transaction.EthereumTransaction
+import com.horizen.account.utils.ZenWeiConverter
+import com.horizen.account.wallet.AccountWallet
+import com.horizen.consensus.ConsensusDataStorage
+import com.horizen.evm.Database
+import com.horizen.fixtures._
+import com.horizen.params.NetworkParams
+import com.horizen.storage.SidechainSecretStorage
+import com.horizen.utils.{ByteArrayWrapper, BytesUtils}
+import com.horizen.{SidechainSettings, SidechainTypes, WalletSettings}
+import org.junit.Assert.{assertEquals, assertTrue}
+import org.junit.{Before, Ignore, Test}
+import org.mockito.{ArgumentMatchers, Mockito}
+import org.scalatestplus.junit.JUnitSuite
+import org.web3j.crypto.{ECKeyPair, Keys}
+import sparkz.core.VersionTag
+import sparkz.core.utils.NetworkTimeProvider
+
+import java.io.{BufferedWriter, FileWriter}
+import java.math.BigInteger
+import java.util.Calendar
+import scala.collection.concurrent.TrieMap
+
+class AccountSidechainNodeViewHolderPerfTest
+    extends JUnitSuite
+      with MockedSidechainNodeViewHolderFixture
+      with EthereumTransactionFixture
+      with StoreFixture
+      with sparkz.core.utils.SparkzEncoding {
+  var historyMock: AccountHistory = _
+  var state: AccountState = _
+  var stateViewMock: AccountStateView = _
+  var wallet: AccountWallet = _
+  var mempool: AccountMemoryPool = _
+
+  implicit val actorSystem: ActorSystem = ActorSystem("sc_nvh_mocked")
+  var mockedNodeViewHolderRef: ActorRef = _
+
+  val mockStateDbNonces = TrieMap[ByteArrayWrapper, BigInteger]()
+
+  @Before
+  def setUp(): Unit = {
+    historyMock = mock[AccountHistory]
+
+    stateViewMock = mock[AccountStateView]
+    Mockito
+      .when(stateViewMock.getBalance(ArgumentMatchers.any[Array[Byte]]))
+      .thenReturn(ZenWeiConverter.MAX_MONEY_IN_WEI) // Has always enough balance
+    Mockito.when(stateViewMock.isEoaAccount(ArgumentMatchers.any[Array[Byte]])).thenReturn(true)
+    Mockito.when(stateViewMock.baseFee).thenReturn(BigInteger.ZERO)
+
+    Mockito.when(stateViewMock.getNonce(ArgumentMatchers.any[Array[Byte]])).thenAnswer { answer =>
+      {
+        mockStateDbNonces.getOrElse(new ByteArrayWrapper(answer.getArgument(0).asInstanceOf[Array[Byte]]), BigInteger.ZERO)
+      }
+    }
+
+    wallet = mock[AccountWallet]
+    Mockito.when(wallet.scanOffchain(ArgumentMatchers.any[SidechainTypes#SCAT])).thenReturn(wallet)
+  }
+
+  @Test
+  @Ignore
+  def txModifyTest(): Unit = {
```

Comment:
Same here, can be good to add explanation comments to easily understand the flow and what we are measuring here without deep diving into the code.

Reply:
Done